credoai/lens/lens.py

```diff
@@ -35,6 +36,27 @@ def __post_init__(self):
         self.evaluator.metadata = self.metadata
         self.metadata["evaluator"] = self.evaluator.name
 
+    @property
+    def identifier(self):
```

Comment:
I spent a while trying to make this more robust. As it is, this seems quite brittle! We're passing an ID that is made up on-the-fly here, to be referenced via a hardcoded set of ID_Cols. Doesn't seem good.

Tried to replace with this:
```
    @property
    def id(self):
        eval_info = self.evaluator.get_info()["metadata"]
        return "~".join(f"{k}:{v}" for k, v in eval_info.items())
```
which makes use of another function we already have to create metadata (I also changed "identifier" => "id")

This results in outputs like this:
`'evaluator:Performance~model_name:credit_RF1_classifier~assessment_dataset_name:UCI-credit-default-test'`

This can't be used later however because the metadata isn't consistent amongst evaluators. However, I'm confused why that _should_ be an issue. Shouldn't we be able to do comparisons evaluator-by-evaluator when they match up? The issue now is coming up because everything is being coerced into one dataframe.

I tried to remove the explicit separation of ID into ID_Cols, but then saw that the ID_Cols were used so that evaluation could be directed to be over the `ref class`, like "model". I was confused about that implementation as well. Once we know that we have the same ID between two lens instances, shouldn't we be able to compare? Why do we need a ref class at all?

Reply:
The reference was split in `ref_type` and `ref` to allow the flexibility of comparing either by model or by whichever dataset (training/assessment) is available. This is to accommodate 2 type of use cases:

1. Many models comparisons: multiple iterations of the same model vs initial case, or different models against a reference one.
2. Multiple dataset comparisons: comparing result when dataset change, over time or due to some form of sampling for example.

If Metadata structure was always constant then I would agree that we can simply pass the `self.evaluator.get_container_info()`, but as it stands I would have to still do extra steps to standardize the output in a similar way to what I am already doing. 

Standardization is also good for results displaying, I think breaking down the id in labelled columns helps user understanding the results, rather then having to read a long and variable id.